contracts/GenArt721CoreV3.sol

```diff
@@ -249,35 +249,35 @@ contract GenArt721CoreV3 is
     ) external returns (uint256 _tokenId) {
         // CHECKS
         require(msg.sender == minterContract, "Must mint from minter contract");
-
+        Project storage project = projects[_projectId];
```

Comment:
Is it possible to get into a state where project is undefined here?

Reply:
Kind of a funny quirk about solidity, mappings can be thought of as "virtually initialized" such that every possible key exists and is mapped to a value whose byte-representation is all zeros (a types default value). So we won't ever have an undefined key.

That being said, what if our minter tried to mint on a project that has not yet been added? When the code below loads `project.invocations` from storage, it will return the default value for `uint24`, which is zero. Similarly, when `project.maxInvocations` is loaded from storage, it will return the default value of zero. Finally, the requirement to have `invocationsBefore < maxInvocations` will fail, preventing the uninitialized project from being minted. Note that the default value for ` bool` is false, so later checks (e.g. `require(project.active,...)` would fail.

Here is a good reference about mappings: https://docs.soliditylang.org/en/develop/types.html?highlight=mapping#mapping-types

Reply:
awesome, thank you for the explanation and the resources! excited to learn more :)